# ==============================================================================
# FIXING COMMIT DIFF
# ==============================================================================
# File: dom/webgpu/CanvasContext.cpp
# Commit: 03d61eb955a9
# Full Hash: 03d61eb955a9ac6e3374ca4c5504edc41cd1aa4a
# Author: sotaro <sotaro.ikeda.g@gmail.com>
# Date: 2023-10-25 16:28:20
# Description:
#   Bug 1860921 - Revert Texture::Destroy() and add Texture::ForceDestroy() r=webgpu-reviewers,nical
#   
#   This is a short term fix. Long term fix needs to be done by Bug 1860958.
#   Presenting WebGPU without readback needs a way to destroy Texture. Then Texture::ForceDestroy() is added for it.
#   
# ==============================================================================

diff -r dc00edd0eac6 -r 03d61eb955a9 dom/webgpu/CanvasContext.cpp
--- a/dom/webgpu/CanvasContext.cpp	Wed Oct 25 09:41:31 2023 +0000
+++ b/dom/webgpu/CanvasContext.cpp	Wed Oct 25 09:43:58 2023 +0000
@@ -201,7 +201,7 @@
   mBridge->SwapChainPresent(mTexture->mId, *mLastRemoteTextureId,
                             *mRemoteTextureOwnerId);
   if (mUseExternalTextureInSwapChain) {
-    mTexture->Destroy();
+    mTexture->ForceDestroy();
     mNewTextureRequested = true;
   }
 }